Implented jelly hunger states

Megafont opened the pull request:
+ Parameters.cs now has hour hunger states and a property to get the current hunger state (which is an enum).
+ I kept the hunger counter, since we need it still to know which state we're in
+ I just left the bar alone for now, but it changes color depending on how hungry he is
+ Fixed a bug in Parameters.cs where Bitter was added to the sum twice, and spicy never was.

Megafont committed:
Implented jelly hunger states

+ Parameters.cs now has hour hunger states and a property to get the current hunger state (which is an enum).
+ I kept the hunger counter, since we need it still to know which state we're in
+ I just left the bar alone for now, but it changes color depending on how hungry he is
+ Fixed a bug in Parameters.cs where Bitter was added to the sum twice, and spicy never was.

CaseyWalshWarder commented:
These changes are good (no bugs or anything), but it's missing the interval between when the jelly can be fed. See the flowchart (https://miro.com/app/board/uXjVNEwPdgg=/?moveToWidget=3458764583989997840&cot=14)

Megafont committed:
Added the eating timer so you can't feed him again for a bit after giving him food

Megafont committed:
Jelly hunger now matches the flowchart fully

+ Changed it so food item is passed into Parameters.cs. This way if different foods have different satiation amounts this can be implemented easily later.
+ Gave jelly a Queue for his stomach in Parameters.cs. First item in is first out. This also lets us know how many food items are in his stomach.

CaseyWalshWarder reviewed (approved):
Good job
